graphql/lessonplan/mutations.ts

```diff
@@ -0,0 +1,219 @@
+
+// Maximum prompt length (in characters)
+const MAX_PROMPT_LENGTH = 10000;
+
+// Maximum number of files
+const MAX_FILES = 10;
+
+// Register the course_subject_enum as a GraphQL enum
+registerEnumType(course_subject_enum, {
+    name: 'CourseSubjectEnum',
+    description: 'The subject of the course',
+});
+
+// Register the school_schooltype_enum as a GraphQL enum
+registerEnumType(school_schooltype_enum, {
+    name: 'SchoolSchoolTypeEnum',
+    description: 'The type of school',
+});
+
+// Create a new enum for expected output fields
+enum LessonPlanField {
+    TITLE = 'title',
+    LEARNING_GOAL = 'learningGoal',
+    AGENDA_EXERCISES = 'agendaExercises',
+    ASSESSMENT = 'assessment',
+    HOMEWORK = 'homework',
+    RESOURCES = 'resources',
+}
+
+// Register the LessonPlanField enum
+registerEnumType(LessonPlanField, {
+    name: 'LessonPlanField',
+    description: 'Fields that can be generated for a lesson plan',
+});
+
+@InputType()
+class GenerateLessonPlanInput {
+    @Field(() => [String], { nullable: true })
+    @ArrayMaxSize(MAX_FILES, { message: `Number of files must not exceed ${MAX_FILES}` })
+    fileUuids?: string[];
+
+    @Field(() => course_subject_enum)
+    subject: course_subject_enum;
+
+    @Field(() => Int)
+    grade: number;
+
+    @Field(() => Int)
+    duration: number;
+
+    @Field(() => pupil_state_enum)
+    state: pupil_state_enum;
+
+    @Field(() => String)
+    @MaxLength(MAX_PROMPT_LENGTH, { message: `Prompt must not exceed ${MAX_PROMPT_LENGTH} characters` })
+    prompt: string;
+
+    @Field(() => [LessonPlanField], { nullable: true })
+    expectedOutputs?: LessonPlanField[];
+
+    @Field(() => school_schooltype_enum)
+    schoolType: school_schooltype_enum;
+
+    @Field(() => String, { nullable: true, defaultValue: 'German' })
+    language?: string;
+}
+
+@ObjectType()
+class LessonPlanOutput {
+    @Field(() => String, { nullable: true })
+    title?: string;
+
+    @Field(() => course_subject_enum)
+    subject: course_subject_enum;
+
+    @Field(() => String)
+    grade: string;
+
+    @Field(() => Int)
+    duration: number;
+
+    @Field(() => String, { nullable: true })
+    learningGoal?: string;
+
+    @Field(() => String, { nullable: true })
+    agendaExercises?: string;
+
+    @Field(() => String, { nullable: true })
+    assessment?: string;
+
+    @Field(() => String, { nullable: true })
+    homework?: string;
+
+    @Field(() => String, { nullable: true })
+    resources?: string;
+}
+
+class LessonPlanGenerationError extends ApolloError {
+    constructor(message: string) {
+        super(message, 'LESSON_PLAN_GENERATION_ERROR');
+
+        Object.defineProperty(this, 'name', { value: 'LessonPlanGenerationError' });
+    }
+}
+
+// Define a type for the state codes based on the enum
+type StateCode = keyof typeof pupil_state_enum;
+
+// Create a mapping object for state codes to full names
+const stateMap: Record<StateCode, string> = {
```

Comment:
Maybe move to https://github.com/corona-school/backend/tree/master/common/util - looks rather generic to me ...